flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/common/CommonExecSink.java

```diff
@@ -197,6 +222,23 @@ public DynamicTableSinkSpec getTableSinkSpec() {
                 .toArray();
     }
 
+    /**
+     * Returns a long[], each long element holds 2 integers, the char field idx and its precision.
+     */
+    private List<Tuple2<Integer, Integer>> getCharFieldIndices(RowType physicalType) {
+        ArrayList<Tuple2<Integer, Integer>> charFieldsAndLengths = new ArrayList<>();
```

Comment:
Use interface for the type

```suggestion
        final List<Tuple2<Integer, Integer>> charFieldsAndLengths = new ArrayList<>();
```